frontend/src/utils/account-send.ts

```diff
@@ -0,0 +1,179 @@
+
+  // Store as much of the pubkey as we can in the remaining bytes. The first 4 characters are `0x04` so we skip those.
+  const pubKeyStart = pubKey.slice(4, remainingBytes * 2 + 4); // Each slot is a half byte meaning remaining bytes must be multiplied by 2
+  const dataToEncrypt = `${address}${advancedModeHalfByte}${usePublicKeyCheckedHalfByte}${pubKeyStart}`;
+  if (dataToEncrypt.length !== 64) {
+    // This should never happen.
+    throw new Error(`Data to encrypt is not the correct length ${dataToEncrypt.length} instead of 64`);
+  }
+
+  return BigNumber.from(`0x${dataToEncrypt}`);
+};
+
+export const encryptAccountData = ({
+  address,
+  advancedMode,
+  usePublicKeyChecked,
+  encryptionCount,
+  viewingKey,
+  pubKey,
+}: EncryptAccountDataArgs) => {
+  const key = keccak256(toUtf8Bytes(`${viewingKey}${encryptionCount}`));
+  const data = buildAccountDataForEncryption({ address, advancedMode, usePublicKeyChecked, pubKey });
+  const encryptedData = data.xor(key);
+  return hexZeroPad(encryptedData.toHexString(), 32);
+};
+
+export const decryptData = ({ viewingKey, encryptionCount, encryptedAddress }: DecryptAccountDataArgs) => {
+  const key = keccak256(toUtf8Bytes(`${viewingKey}${encryptionCount}`));
+
+  const decryptedData = BigNumber.from(encryptedAddress).xor(key);
+  const hexData = decryptedData.toHexString();
+
+  const paritalPubKey = hexData.slice(44);
+  const advancedMode = hexData.slice(42, 43);
+  const usePublicKeyChecked = hexData.slice(43, 44);
+  return {
+    advancedMode,
+    usePublicKeyChecked,
+    address: hexData.slice(0, 42),
+    pubKey: paritalPubKey,
+  };
+};
+
+export const storeSend = async ({
+  recipientAddress,
+  chainId,
+  advancedMode,
+  usePublicKeyChecked,
+  viewingKey,
+  amount,
+  tokenAddress,
+  txHash,
+  userAddress,
+  provider,
+  pubKey,
+}: StoreSendArgs) => {
+  // Send history is scoped by chain
+  const key = `${LOCALFORAGE_ACCOUNT_SEND_KEY}-${userAddress}-${chainId}`;
+  const count =
+    ((await localforage.getItem(`${LOCALFORAGE_ACCOUNT_SEND_KEY}-count-${userAddress}-${chainId}`)) as number) || 0;
+  const checksummedRecipientAddress = await toAddress(recipientAddress, provider);
+  const encryptedData = encryptAccountData({
+    address: checksummedRecipientAddress,
+    advancedMode,
+    usePublicKeyChecked,
+    encryptionCount: count,
+    viewingKey,
+    pubKey,
+  });
+  const values = ((await localforage.getItem(key)) as AccountSendData[]) || [];
+  await localforage.setItem(key, [
+    ...values,
+    {
+      encryptedAddress: encryptedData,
+      amount,
+      tokenAddress,
+      dateSent: new Date(),
+      txHash,
+    },
+  ]);
+  await localforage.setItem(`${LOCALFORAGE_ACCOUNT_SEND_KEY}-count-${userAddress}-${chainId}`, count + 1);
+};
+
+export const fetchAccountSends = async ({ address, viewingKey, chainId }: FetchAccountSendArgs) => {
+  const key = `${LOCALFORAGE_ACCOUNT_SEND_KEY}-${address}-${chainId}`;
```

Comment:
Same comment about renaming this var

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/945e85030f56ac832dd39f2fc4d57effd3554c17